cpp/include/raft/selection/col_wise_sort.cuh

```diff
@@ -0,0 +1,346 @@
+                    INST_BLOCK_SORT(in, sortedKeys, out, n_rows, n_columns, 512, 8, stream);
+                else if (n_columns > 4096 && n_columns <= 6144)
+                    INST_BLOCK_SORT(in, sortedKeys, out, n_rows, n_columns, 512, 12, stream);
+                else
+                    INST_BLOCK_SORT(in, sortedKeys, out, n_rows, n_columns, 1024, 12, stream);
+            } else if (n_columns <= (1 << 18) && n_rows > 1) {
+                // device Segmented radix sort
+                // 2^18 column cap to restrict size of workspace ~512 MB
+                // will give better perf than below deviceWide Sort for even larger dims
+                int numSegments = n_rows + 1;
+
+                // need auxillary storage: cub sorting + keys (if user not passing) +
+                // staging for values out + segment partition
+                if (workspaceSize == 0 || !workspacePtr) {
+                    OutType* tmpValIn    = nullptr;
+                    int* tmpOffsetBuffer = nullptr;
+
+                    // first call is to get size of workspace
+                    CUDA_CHECK(cub::DeviceSegmentedRadixSort::SortPairs(workspacePtr,
+                                                                        workspaceSize,
+                                                                        in,
+                                                                        sortedKeys,
+                                                                        tmpValIn,
+                                                                        out,
+                                                                        totalElements,
+                                                                        numSegments,
+                                                                        tmpOffsetBuffer,
+                                                                        tmpOffsetBuffer + 1));
+                    bAllocWorkspace = true;
+                    // more staging space for temp output of keys
+                    if (!sortedKeys)
+                        workspaceSize += raft::alignTo(sizeof(InType) * (size_t)totalElements, memAlignWidth);
+
+                    // value in KV pair need to be passed in, out buffer is separate
+                    workspaceSize += raft::alignTo(sizeof(OutType) * (size_t)totalElements, memAlignWidth);
+
+                    // for segment offsets
+                    workspaceSize += raft::alignTo(sizeof(int) * (size_t)numSegments, memAlignWidth);
+                } else {
+                    size_t workspaceOffset = 0;
+
+                    if (!sortedKeys) {
+                        sortedKeys      = reinterpret_cast<InType*>(workspacePtr);
+                        workspaceOffset = raft::alignTo(sizeof(InType) * (size_t)totalElements, memAlignWidth);
+                        workspacePtr    = (void*)((size_t)workspacePtr + workspaceOffset);
```

Comment:
Any particular reason for a C-style cast as opposed to a `reinterpret_cast` here?